src/Lookup/RestrictedEntityLookup.php

```diff
@@ -129,4 +140,11 @@ public function entityHasBeenAccessed( EntityId $entityId ) {
 		return array_key_exists( $entityIdSerialization, $this->entitiesAccessed );
 	}
 
+	/**
+	 * @param string[] $disabledEntityTypes
+	 */
+	public function setDisabledEntityTypes( array $disabledEntityTypes ) {
```

Comment:
I disagree with the idea to have a setter for this. It introduces an issue I find quite problematic: the behavior of this service can be changed any time by any code that is allowed to use this service. This should be a constructor argument.

When doing so it also becomes a little more obvious that Jeroen's suggestion to introduce an other decorator (and possibly rename this one here) might be a cleaner solution.

Reply:
I wanted to go with constructor but Daniel suggested to go with setter specially because it has a sensible (empty array) default. FWIW, How much does it matter at the end?

Reply:
As explained I believe it matters a lot.